From 9c16f42fa584ecb7e6c27124a0bd76fcca182bdb Mon Sep 17 00:00:00 2001
From: Daniel Templeton <templedf@apache.org>
Date: Tue, 15 Nov 2016 13:01:57 -0800
Subject: [PATCH 2039/2848] YARN-5736 Addendum. Fixes segfault due to
 unterminated string. (Contributed by Miklos
 Szegedi via Daniel Templeton)

(cherry picked from commit 264ddb1)

Change-Id: I4b31313c74b5570e4ffa4ca7437717bb5a1bb709
---
 .../native/container-executor/impl/configuration.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/configuration.c b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/configuration.c
index ebbcf34..cb12ed4 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/configuration.c
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/configuration.c
@@ -151,6 +151,7 @@ char* trim(char* input)
     }
 
     strncpy(ret, val_begin, val_end - val_begin);
+    ret[val_end - val_begin] = '\0';
     return ret;
 }
 
-- 
1.7.9.5

